src/pfe/file-watcher/server/test/unit-test/tests/actions.module.test.ts

```diff
@@ -206,4 +259,435 @@ export function actionsTestModule(): void {
+                "args": {
+                    projectType: "liberty",
+                    projectID: "javaMicroProfileTemplate",
+                    location: path.join(process.env.CW_WORKSPACE, "javaMicroProfileTemplate")
+                },
+                "result": "failed",
+                "error": "Missing required file"
+            },
+            "combo7": {
+                "description": "microprofile project with root user dockerfile-build",
+                "args": {
+                    projectType: "liberty",
+                    projectID: "javaMicroProfileTemplate",
+                    location: path.join(process.env.CW_WORKSPACE, "javaMicroProfileTemplate")
+                },
+                "result": "failed",
+                "error": "Migration may required to run project container as a non-root user"
+            },
+            "combo8": {
+                "description": "microprofile project with bad inner most pom.xml elements",
+                "args": {
+                    projectType: "liberty",
+                    projectID: "javaMicroProfileTemplate",
+                    location: path.join(process.env.CW_WORKSPACE, "javaMicroProfileTemplate")
+                },
+                "result": "failed",
+                "error": ["Missing Liberty parent POM groupId", "Missing Liberty parent POM artifactId", "Missing profile activation", "liberty-maven-plugin extensions not enabled", "liberty-maven-plugin looseApplication is not enabled"],
+            },
+            "combo9": {
+                "description": "microprofile project with bad pom.xml due to missing liberty maven plugin",
+                "args": {
+                    projectType: "liberty",
+                    projectID: "javaMicroProfileTemplate",
+                    location: path.join(process.env.CW_WORKSPACE, "javaMicroProfileTemplate")
+                },
+                "result": "failed",
+                "error": "Missing liberty-maven-plugin configuration"
+            },
+            "combo10": {
+                "description": "microprofile project with bad pom.xml due to bad profile",
+                "args": {
+                    projectType: "liberty",
+                    projectID: "javaMicroProfileTemplate",
+                    location: path.join(process.env.CW_WORKSPACE, "javaMicroProfileTemplate")
+                },
+                "result": "failed",
+                "error": "A maven profile named 'microclimate' was not found."
+            },
+            "combo11": {
+                "description": "microprofile project with bad pom.xml syntax",
+                "args": {
+                    projectType: "liberty",
+                    projectID: "javaMicroProfileTemplate",
+                    location: path.join(process.env.CW_WORKSPACE, "javaMicroProfileTemplate")
+                },
+                "result": "failed",
+                "error": "POM parsing error"
```

Comment:
I think you missed out on one Liberty validation: `libertyProject.validator.parentPOM.label` which is in the first catch block of `LibertyValidator.validatePOM`

Adding this probably means having another copy of a garbage pom i think? I shall leave this upto you

Reply:
Added.